packages/block-library/src/navigation/style.scss

```diff
@@ -728,28 +726,17 @@ button.wp-block-navigation-item__content {
 	}
 }
 
-.is-menu-open .wp-block-navigation__responsive-close,
-.is-menu-open .wp-block-navigation__responsive-dialog,
-.is-menu-open .wp-block-navigation__responsive-container-content {
+.wp-block-navigation__responsive-container:is(.is-menu-open, [open])
+:is(.wp-block-navigation__responsive-close, .wp-block-navigation__responsive-dialog, .wp-block-navigation__responsive-container-content) {
 	box-sizing: border-box;
 }
 
 .wp-block-navigation__responsive-dialog {
 	position: relative;
 }
 
-// Adjust open dialog top margin when admin-bar is visible.
-// Needs to be scoped to .is-menu-open, or it will shift the position of any other navigations that may be present.
-.has-modal-open .admin-bar .is-menu-open .wp-block-navigation__responsive-dialog {
-	margin-top: $admin-bar-height-big;
-
-	// Handle smaller admin-bar.
-	@include break-medium() {
-		margin-top: $admin-bar-height;
-	}
-}
```

Comment:
The overlay (`dialog`) is now in the top-layer and the admin bar is unable to appear in front of it so these styles are irrelevant.